src/datasets/arrow_dataset.py

```diff
@@ -5615,6 +5615,25 @@ def iter_random_indices():
     return concatenated_datasets.select(indices, **kwargs)
 
 
+def _split_by_node_map_style_dataset(dataset: Dataset, rank: int, world_size: int) -> Dataset:
+    """
+    Split a dataset for the node at rank `rank` in a pool of nodes of size `world_size`.
+    Each node is assigned a chunk of data, e.g. rank 0 is given the first chunk of the dataset.
+
+    Args:
+        dataset ([`Dataset`]):
+            The dataset to split by node.
+        rank (`int`):
+            Rank of the current node.
+        world_size (`int`):
+            Total number of nodes.
+
+    Returns:
+        [`Dataset`]: The dataset to be used on the node at rank `rank`.
+    """
+    return dataset.shard(num_shards=world_size, index=rank, contiguous=True)
```

Comment:
The iterable version does return a dataset of contiguous data if possible to get the best performance. Here if we don't return contiguous data the performance would also get worse because of the indices mapping

Reply:
I'd assume TFDS does `contiguous=False` by default to diversify the shards, but indeed this is not ideal for us due to the indices mapping, so a note sounds good to me.